### ISSUE TITLE
Setup bluebird and/or wikipedia benchmarks for promise performance

### ISSUE THREAD

**Issue by bmeurer** [nodejs/benchmarking#188]
Specifically we're interested in the difference between regular runs with these workloads and runs where we have `async_hooks` enabled (see #181 for background).

**Comment by bmeurer** [comment]
Ok, I managed to run the aforementioned benchmarks today with and without `async_hooks`, and the results show a significant slow-down, even with just an empty `init` hook.

![Results for Node 9.3.0](https://raw.githubusercontent.com/bmeurer/async-hooks-performance-impact/master/results-node-9.3.0.png)

The full results and benchmarks are available at https://github.com/bmeurer/async-hooks-performance-impact

**Comment by AndreasMadsen** [comment]
@bmeurer By bluebird, do you mean with https://github.com/petkaantonov/bluebird/pull/1472 applied?

**Comment by bmeurer** [comment]
No it's the bluebird benchmarks running with native promises.

**Comment by ak239** [comment]
It looks like to implement async_hooks for promises V8 enters C++ code and then go back into JavaScript land to call actual callback. Should we somehow migrate promise hooks from C++ builtin to TF builtin to get a performance boost?

And probably it is not best place for this question but what is main use case for async_hooks? Is it for some kind of async context tracking?

**Comment by bmeurer** [comment]
@gsathya already suggested to move the promise hooks to JS and eventually spec them. This would help to reduce the C++ overhead, but so far I don't think that's the only/main bottleneck.

**Comment by mhdawson** [comment]
Discussed, last meeting, nothing for TSC to decide at this point.  Let async hooks team address, removing tag for now.

**Comment by ofrobots** [comment]
Seems like @nodejs/async_hooks hasn't been mentioned here yet. Doing so.

**Comment by Trott** [comment]
Seems like the `tsc-agenda` label should be removed? If I'm wrong about that, by all means, add it again.

**Comment by mhdawson** [comment]
@Trott thanks, I had removed manually from the last agenda, but forgot to remove the tag.

**Comment by mhdawson** [comment]
@bmeurer since we have no other promise centric benchmarks I'm wondering if running these nightly would be better than nothing?

**Comment by bmeurer** [comment]
@mhdawson Makes sense. I've created versions of these benchmarks, see https://github.com/v8/promise-performance-tests, which we use for V8 internally to track `Promise` and `async/await` performance. Shouldn't be hard to setup on Node infrastructure as well (and run with `async_hook` on and off).
